spec/unit/request_spec.rb

```diff
@@ -618,6 +618,22 @@
 
 
   describe "logging" do
+    it "logs custom user agent" do
+      log = RestClient.log = []
+      RestClient.user_agent = "Foo"
+      RestClient::Request.new(:method => :get, :url => 'http://url').log_request
+      RestClient.user_agent = nil
```

Comment:
A subtle point maybe worth testing: is this affected by value at moment of `RestClient::Request` construction, or request execution?
```ruby
      RestClient.user_agent = "Foo"
      req = RestClient::Request.new(:method => :get, :url => 'http://url')
      RestClient.user_agent = "Bar"
      req.log_request
      RestClient.user_agent = nil
```
`default_headers` called from `make_headers` called from initialize, so I guess construction time?
Construction time is less surprising IMHO :+1:, but I think `RestClient.proxy` is used when request is made, and maybe consistency is better.  I don't have a strong opinion here.